src/default-request-executor.js

```diff
@@ -0,0 +1,108 @@
+/*!
+ * Copyright (c) 2018-present, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+const RequestExecutor = require('./request-executor');
+const deepCopy = require('deep-copy');
+class DefaultOktaRequestExecutor extends RequestExecutor {
+  constructor(config = {}) {
+    super();
+    this.maxElapsedTime = config.maxElapsedTime || 60000;
+    this.rateLimitRandomOffsetMin = config.rateLimitRandomOffsetMin || 1000;
+    this.rateLimitRandomOffsetMax = config.rateLimitRandomOffsetMax || 5000;
+    this.retryCountHeader = 'X-Okta-Retry-Count';
+    this.retryForHeader = 'X-Okta-Retry-For';
+  }
+
+  buildRetryRequest(request, response) {
+    const newRequest = deepCopy(request);
+    const requestId = this.getOktaRequestId(response);
+    if (!request.startTime) {
+      newRequest.startTime = new Date();
+    }
+    if (!newRequest.headers) {
+      newRequest.headers = {};
+    }
+    if (!newRequest.headers[this.retryForHeader]) {
+      newRequest.headers[this.retryForHeader] = requestId;
+    }
+    newRequest.headers[this.retryCountHeader] =
+      newRequest.headers[this.retryCountHeader] ?
+      newRequest.headers[this.retryCountHeader] + 1 : 1;
+    return newRequest;
+  }
+
+  canRetryRequest(response) {
+    // Validate that we don't have duplicate headers, see OKTA-112507
+    // Duplicate headers are returned by fetch as a comma separated list.
+    const retryHeader = this.getRateLimitReset(response);
+    return !!(retryHeader && retryHeader.indexOf(',') === - 1);
```

Comment:
Just to double check, when OKTA-112507 is resolved, we can remove `&& retryHeader.indexOf(',') === - 1`?

Reply:
IMO we should keep it, it is possible this could regress, or some other proxy could add additional headers (unlikely though)